### PR TITLE
`gq` range when cursor is in the middle of a line.

### LINKED ISSUE #4590: gq fails when cursor is in middle of a line
For situation like this:
```
    // pine |apple 
    // SpongeBob Patrick
```
With `gqj`, we should get result like this:
```
    |// pine apple SpongeBob Patrick
```
But currently we get:
```
|    // pine apple
// SpongeBobPatrick
```

### PR DESCRIPTION
**Pull request by ldm0**
**Bugs:**
Incorrect behaviour when cursor is in the middle of a line.(fixed by expanding start and end.)

**Small Refactor:**
Indentation should be got in paragraph reflow rather than being passed in as a parameter.

Fixes #4590

### CODE REVIEW

File: test/mode/modeNormal.test.ts [hunk 1]
```diff
@@ -1402,7 +1402,7 @@ suite('Mode Normal', () => {
     start: [
       '//    We choose to write a vim extension, not because it is easy, but because it is hard|.',
     ],
-    keysPressed: 'Vgq',
+    keysPressed: 'gq$',
```
Comment (hao-flexport) [inline test/mode/modeNormal.test.ts]: I think we should both cover `Vgq` and `gp$`.

File: src/actions/operator.ts [hunk 2]
```diff
@@ -1094,21 +1096,22 @@ class ActionVisualReflowParagraph extends BaseOperator {
...
 
     vimState.recordedState.transformations.push({
       type: 'replaceText',
       text: textToReflow,
       start: start,
       end: end,
-      // Move cursor to front of line to realign the view
-      diff: PositionDiff.newBOLDiff(),
+      // Move cursor to front of lowest line to realign the view
+      diff: PositionDiff.newBOLDiff(textToReflow.split('\n').length - 1),
```
Comment (hao-flexport) [inline src/actions/operator.ts]: Is it safe to just split by `\n`? I mean can this handle `\r\n`.
Hope I am just over thinking.

Comment (ldm0) [reply src/actions/operator.ts]: Yes. `"\n".split('\n')` and `"\r\n".split('\n')` will return the same result. You can check that right now by pressing f12. :-P

File: test/mode/modeNormal.test.ts (hunk 1 above)
Comment (ldm0) [reply test/mode/modeNormal.test.ts]: I also want to cover the "Vgq". But I don't have enough time to work on the bug of `V`. Current implementation of `V` is just just kind of workaround and it's result is not like what we expected. If we insist on testing this `Vgq`, the more important thing will be postponed.

Comment (hao-flexport) [reply test/mode/modeNormal.test.ts]: I think you can keep previous tests, declaring the wrong behaviors with comment/explanation, and create your new ones.
Hope other members would not oppose.

File: src/actions/operator.ts [hunk 3]
```diff
@@ -1086,18 +1088,19 @@ class ActionVisualReflowParagraph extends BaseOperator {
...
```
Comment (J-Fields) [inline src/actions/operator.ts:1103]: This doesn't work when the cursor is already at the bottom (`gq5k`, for instance)

Comment (ldm0) [reply src/actions/operator.ts:1103]: @J-Fields Ah, that's really a problem. :-/ I'll revert it.

**Pull request merged**